### PR TITLE
Updated the meshkit error in meshery_application_handler

### LINKED ISSUE #3281: [Chore] Update Handlers to use new Meshkit Errors #3
Note: This is a child issue of #3275 
#### Current Behavior
Meshery Server (and its other components, like Adapters) use a centralized library, MeshKit, to provide uniform utilities (e.g. raising and logging errors). Currently, Meshery Server uses a now antiquated style of error logging with logrus.

#### Desired Behavior
Ideally, all use of logrus is replaced with use of MeshKit-style error logging.

#### Implementation
Meshery Handlers needs to be:
 - Updated to use meshkit errors. 
 - Switch to handler Log instead of `logrus`

Handlers to update:

- [x] user handler
- [x]  mesh ops handler
- [ ]  meshery application handler

For reference, you can checkout  the [`grafana handler`](https://github.com/layer5io/meshery/blob/master/handlers/grafana_handlers.go) and [`prometheus handler`](https://github.com/layer5io/meshery/blob/master/handlers/prometheus_handlers.go) or the [docs](https://docs.meshery.io/project/contributing-error).
Define all Error functions for all the handlers to be written in the `handlers/error.go` file.

#### Mockups

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Updated the mesh kit error for the meshery_application_handler

This PR fixes #3281 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: handlers/error.go [hunk 1]
```diff
@@ -101,7 +104,7 @@ func ErrUnmarshal(err error, obj string) error {
 }
 
 func ErrEncoding(err error, obj string) error {
-	return errors.New(ErrEncodingCode, errors.Alert, []string{"Error encoding the : " + obj}, []string{err.Error()}, []string{"Unable to decode the : " + obj}, []string{})
+	return errors.New(ErrEncodingCode, errors.Alert, []string{"Error encoding the : " + obj}, []string{err.Error()}, []string{"Unable to encode the : " + obj}, []string{})
```
Comment (dhruv0000) [inline handlers/error.go:107]: Instead to appending the string, lets move to spperating then by `,`, like
```
return errors.New(ErrEncodingCode, errors.Alert, []string{"Error encoding the object", obj}, []string{err.Error()}, []string{"Unable to encode the object ", obj}, []string{})
```

This way the error utility tool would be able to detect and export the string for the docs, while also being able to see the object in meshery server.

`ErrUnmarshal` and `ErrMarshal` would also need some rework.

File: handlers/meshery_application_handler.go [hunk 2]
```diff
@@ -51,14 +51,17 @@ func (h *Handler) handleApplicationPOST(
 
 	var parsedBody *MesheryApplicationRequestBody
 	if err := json.NewDecoder(r.Body).Decode(&parsedBody); err != nil {
-		rw.WriteHeader(http.StatusBadRequest)
-		fmt.Fprintf(rw, "failed to read request body: %s", err)
+		h.log.Error(ErrRetrieveData(err))
+		http.Error(rw, ErrRetrieveData(err).Error(), http.StatusBadRequest)
+		// rw.WriteHeader(http.StatusBadRequest)
```
Comment (dhruv0000) [inline handlers/meshery_application_handler.go:56]: Looks Good. Nice Work :+1:

**Review by dhruv0000 — CHANGES_REQUESTED** [review]

File: handlers/error.go (hunk 1 above)
Comment (sayantan1413) [reply handlers/error.go:107]: Yeah okay I will do the rework.

File: handlers/error.go [hunk 3]
```diff
@@ -187,3 +190,15 @@ func ErrApplyChange(err error) error {
 func ErrRetrieveMeshData(err error) error {
 	return errors.New(ErrRetrieveMeshDataCode, errors.Alert, []string{"Error getting operations for the mesh", "Error getting service mesh name"}, []string{err.Error()}, []string{"unable to retrieve the requested data"}, []string{})
 }
+
+func ErrApplicationFailure(err error, obj string) error {
+	return errors.New(ErrApplicationFailureCode, errors.Alert, []string{"failed to ", obj, "the application"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrDecoding(err error, obj string) error {
+	return errors.New(ErrDecodingCode, errors.Alert, []string{"Error decoding the : ", obj}, []string{err.Error()}, []string{"Unable to decode the : ", obj}, []string{})
```
Comment (kumarabd) [inline handlers/error.go:199]: @sayantan1413 Uve filled `"Unable to decode the : "` in probable cause, lets have It blank.

**Comment by kumarabd** [comment]
Rest LGTM

**Comment by sayantan1413** [comment]
Hey @kumarabd I have corrected it now. Can you have a look?

**Review by kumarabd — APPROVED** [review]

**Pull request merged**